addon/mixins/scoped-query-cache.js

```diff
@@ -73,5 +76,21 @@ export default Ember.Mixin.create({
 
       return results;
     });
+  },
+
+  scheduleCacheDecay(delay = CACHE_DECAY_TIMEOUT, callback) {
+    this.decayId = setTimeout(() => {
+      this.cache = this._createCache();
+
+      callback && typeof callback === 'function' && callback();
```

Comment:
Agree with Sang's thought on the design. Predicate-based caching would be more performant, and avoid adding async code to consuming code.